### PR TITLE
feature(badge): badges api integration

### PR DESCRIPTION
**Pull request by mthinh**
# What ?
- Implement fetching badges 
- Implement filter badges by by collection, location, distance
- Implement claim badges

### CODE REVIEW

**Review by roncoderpush — CHANGES_REQUESTED** [review]

File: lib/core/presentation/widgets/poap/poap_item.dart [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+  });
+
+  final TokenMetadata? tokenMetadata;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      width: 72,
+      height: 72,
+      decoration: ShapeDecoration(
```
Comment (RonTran636) [inline lib/core/presentation/widgets/poap/poap_item.dart:149]: Use flutter screen utils for large raw size. Ex 72.w

File: lib/core/presentation/widgets/poap/poap_item.dart [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+    final distanceInKm = (badge.distance ?? 0) / 1000;
+    String displayDistance;
+    if (distanceInKm >= 1) {
+      displayDistance = NumberFormat.compact().format(distanceInKm);
+    } else {
+      displayDistance = NumberFormat('##.##').format(badge.distance ?? 0);
+    }
+    return SizedBox(
+      height: 30,
+      child: Row(
```
Comment (RonTran636) [inline lib/core/presentation/widgets/poap/poap_item.dart:228]: replace it with 30.h

File: lib/core/service/badge/badge_service.dart [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+          BadgeLocation.myLocation(
+            geoPoint: GeoPoint(
+              lat: position.latitude,
+              lng: position.longitude,
+            ),
+          ),
+        );
+      }
+    } catch (error) {}
+  }
```
Comment (RonTran636) [inline lib/core/service/badge/badge_service.dart:60]: avoid an empty catch. If no action is required, at least put log here so we know if it goes error

File: lib/core/service/shake/shake_service.dart [hunk 4]
```diff
@@ -37,47 +33,45 @@ class ShakeService {
...
       barrierDismissible: false,
-      builder: (BuildContext context) {
+      builder: (context) {
         return Dialog(
           child: StatefulBuilder(
             builder: (context, setState) {
               return Container(
-                padding: EdgeInsets.all(16.0),
+                padding: const EdgeInsets.all(16),
                 child: Column(
```
Comment (RonTran636) [inline lib/core/service/shake/shake_service.dart:46]: use Sizing class or 16.w instead

File: lib/core/service/shake/shake_service.dart [hunk 5]
```diff
@@ -37,47 +33,45 @@ class ShakeService {
...
                   mainAxisSize: MainAxisSize.min,
                   crossAxisAlignment: CrossAxisAlignment.start,
                   children: [
-                    Text(
+                    const Text(
                       'Debug info',
-                      style: TextStyle(
-                          fontSize: 24.0, fontWeight: FontWeight.bold),
+                      style: TextStyle(fontSize: 24, fontWeight: FontWeight.bold),
                     ),
```
Comment (RonTran636) [inline lib/core/service/shake/shake_service.dart:53]: for fontsize use `.sp`

File: lib/core/service/shake/shake_service.dart [hunk 6]
```diff
@@ -37,47 +33,45 @@ class ShakeService {
...
-                    SizedBox(height: 16.0),
+                    const SizedBox(height: 16),
                     Text(
```
Comment (RonTran636) [inline lib/core/service/shake/shake_service.dart:55]: Same as above, should replace with `.w` all place in this class

**Review by RonTran636 — CHANGES_REQUESTED** [review]

File: lib/core/presentation/pages/poap/popap_listing_page.dart [hunk 7]
```diff
@@ -1,86 +1,86 @@
...
+import 'package:app/core/presentation/pages/poap/views/poap_listing_view.dart';
 import 'package:auto_route/auto_route.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
 
 @RoutePage()
-class PoapListingPage extends StatelessWidget {
-  PoapListingPage({super.key});
+class PoapListingPage extends StatefulWidget {
+  const PoapListingPage({super.key});
```
[the remark this answers is not in the document]
Comment (mthinh) [reply lib/core/presentation/pages/poap/popap_listing_page.dart:12]: Oh, it's used to be stateful, I did removed some logic, let be convert it back to stateless

File: lib/core/presentation/widgets/poap/poap_claim_builder.dart [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    required this.builder,
+  });
+  final badge_entities.Badge badge;
+  final Widget Function(BuildContext context, ClaimPoapState state, bool locationEnabled) builder;
+
+  @override
+  Widget build(BuildContext context) {
+    return _PoapClaimBuilderView(
+      builder: builder,
+    );
```
[the remark this answers is not in the document]
Comment (mthinh) [reply lib/core/presentation/widgets/poap/poap_claim_builder.dart:21]: Used to wrap the parent with BlocProvider, let me merge it.

File: lib/core/presentation/widgets/poap/poap_item.dart (hunk 1 above)
Comment (mthinh) [reply lib/core/presentation/widgets/poap/poap_item.dart:149]: 👍

File: lib/core/service/shake/shake_service.dart (hunk 5 above)
Comment (mthinh) [reply lib/core/service/shake/shake_service.dart:53]: 👍

File: lib/core/domain/user/entities/user.dart [hunk 9]
```diff
@@ -4,70 +4,36 @@ import 'package:app/core/domain/common/entities/common.dart';
...
-  final String? handleTwitter;
-  final String? handleInstagram;
-  final String? handleFacebook;
-  final String? handleLinkedin;
-
-  final List<String>? wallets;
-  final String? walletCustodial;
 
   User({
     this.id,
```
[the remark this answers is not in the document]
Comment (mthinh) [reply lib/core/domain/user/entities/user.dart:39]: It depends on your specific requirements, but generally "User" is fine if you prefer shorter, succinct names, while "UserEntity" can be used to specify that you are referring to an actual database/entity object. Consider consistency and clarity in your naming conventions.

I asked ChatGPT haha, maybe for consistency in the current codebase, User is just enough

File: lib/core/service/shake/shake_service.dart (hunk 4 above)
Comment (mthinh) [reply lib/core/service/shake/shake_service.dart:46]: this dialog only for testing

**Comment by mthinh** [comment]
@roncoderpush shake service dialog only for testing so I will not apply sizing. Other comments have been fixed. Please help to recheck. Thank you

**Review by sieuhuflit — APPROVED** [review]
Overall look good on my side 💯

**Comment by roncoderpush** [comment]
@sieuhuflit @mthinh LGTM, nice work

**Comment by roncoderpush** [comment]
@sieuhuflit it should be merge by now, can you do the honor :D

**Pull request merged**